lib/daru/io/importers/redis.rb

```diff
@@ -0,0 +1,66 @@
+require 'daru/io/importers/linkages/redis'
+require 'json'
+
+module Daru
+  module IO
+    module Importers
+      class Redis
+        def initialize(connection={}, *keys)
+          @client = get_client(connection)
+          @keys   = choose_keys(*keys)
+        end
+
+        def load
+          vals = @keys.map { |key| ::JSON.parse(@client.get(key)) }
+
+          case ::JSON.parse(@client.get(@keys[0]))
+          when Array
+            case ::JSON.parse(@client.get(@keys[0]))[0]
+            when Hash
+              # Array of hashes
+              # key a :
+              # [
+              #   { x: 1, y: 2 },
+              #   { x: 3, y: 4 }
+              # ]
+              Daru::DataFrame.new vals.flatten
+            else
+              # Hash containing Array
+              # key a :
+              # {
+              #   x: [1,2,3,4]
+              #   y: [5,6,7,8]
+              # }
+              Daru::DataFrame.rows vals.transpose, order: @keys
+            end
+          when Hash
+            # Array containing Hash
+            # [
+            #   key a: { x: 1, y: 2 }
+            #   key b: { x: 3, y: 4 }
+            # ]
+            Daru::DataFrame.new vals.flatten, index: @keys
+          end
+        end
+
+        private
+
+        def choose_keys(*keys)
+          if keys.count.zero?
+            @client.keys
```

Comment:
Points noted. I myself got an input from a friend regarding how they use redis primarily for caching data timestamp-wise. This was the inspiration behind my assumption that all / selected keys will be 'similar'. I was thinking of also adding a regexp-like match on keys, so that only they keys that look like a timestamp can be collected - but #scan impressive too.